gwpy/table/io/snax.py

```diff
@@ -33,26 +33,46 @@
 @read_with_columns
 @read_with_selection
 @with_read_hdf5
-def table_from_file(source, channel):
+def table_from_file(source, channels=None):
     """Read an `EventTable` from a SNAX HDF5 file
 
     Parameters
     ----------
     source : `h5py.File`
         the file path of open `h5py` object from which to read the data
 
-    channel : `str`
-        the channel to read data from
+    channels : `str` or `list`
+        the channel(s) to read data from
 
     Returns
     -------
     table : `~gwpy.table.EventTable`
     """
-    return vstack(
-        list(map(read_table_hdf5, source[channel].values())),
-        join_type="exact",
-        metadata_conflicts="error",
-    )
+    # format channels appropriately
+    if isinstance(channels, str):
+        channels = [channels]
+
+    # only query channels contained in file
+    # if channels not specified, load all of them
+    if channels is None:
+        channels = source.keys()
+    else:
+        channels = set(channels) & set(source.keys())
```

Comment:
I'm not a fan of silently ignoring channels that were explicitly requested but aren't included in the file. I would rather this be an error by default.

You could consider adding a keyword `on_missing="error"` to allow users to downgrade to a warning if their channel is missing. Something like this:

```suggestion
        channels = set(channels)
        found = set(source.keys())
        missing = channels - found
        msg = "requested channels not found in SNAX file: '{}'".format(
            "', '".join(missing),
        )
        if missing and on_missing == "error":
            raise ValueError(msg)
        if missing and on_missing == "warn":
            warnings.warn(msg)
        channels = channels & found
```

Reply:
Thanks @duncanmmacleod, I agree with all that and I'll add this in.

As an aside, after submitting this PR I realized how much space the 'channel' column takes in memory (somewhere between a `U30` and a `U40` which takes up around 150-200B per row). It's probably fine for typical use cases but would be expensive when you're reading many tables/channels which is typical for workflows like iDQ/hveto. I have an implementation that should get around that with some more work from the user (`compact`, False by default) by storing an integer representation of the channel and a `channel_map` in the table's metadata to map between the integer representation (really just `hash(channel`) and the full channel name).

Switching back involves something like:
```
table = table.group_by('channel')
for key, group in zip(table.groups.keys, table.groups):
    group['channel'] = table.meta['channel_map'][key['channel']]
```

First, how do you feel about something like that? That implementation is sufficient for my use case but I don't know if that's necessarily the best solution to expose in general, even if a user would have to explicitly toggle it on. Second, if you're favorable to that, would you want that as part of this PR or in a follow-up PR?

Reply:
So in your case, you need the `channel` information for each row, but are just trying to cut down on memory overhead? Or could you live without the channel column entirely?

Reply:
That's right, the former. I would need the channel information if I'm requesting multiple channels since I'd like to find coincidences between a target channel and other channels. For a typical iDQ run, I'm requesting O(100) channels for O(day) stretch in time which can be problematic in terms of memory usage in some cases, but this would likely not be a concern for other typical use cases.

Reply:
I think I'm ok with the opt-in `compact` representation you have proposed - please implement it, then I can take a proper look and see what I really think.

Reply:
@duncanmmacleod, just added this in the latest commit along with the relevant tests.